src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/restapi/model/IndividualNeedsScores.kt

```diff
@@ -142,11 +147,15 @@ data class IndividualRelationshipScores(
       (aggressiveControllingBehaviour ?: 0)
   }
 
-  fun overallRelationshipScore() = when (totalScore()) {
-    0 -> null
-    in 0..1 -> 0
-    in 2..4 -> 1
-    in 5..8 -> 2
-    else -> null
+  fun overallRelationshipScore(): Int? {
+    if (!areAllValuesPresent()) return null
+
+    return when (totalScore()) {
+      0 -> null
+      in 0..1 -> 0
+      in 2..4 -> 1
+      in 5..8 -> 2
+      else -> null
+    }
```

Comment:
return when (val score = totalScore()) {
    in 1..1 -> 0
    in 2..4 -> 1
    in 5..8 -> 2
    else -> null
}

This way we could lose the 0 check at the start

Reply:
You're right, the zero check is surplus to requirements now. Will remove.